src/main/java/lotto/RawExpression.java

```diff
@@ -0,0 +1,35 @@
+package lotto;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RawExpression {
+
+    private String customDelimiter;
+    private String expression;
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.)\n(.*)";
+    private static final Integer MATCH_GROUP_DELIMITER = 1;
+    private static final Integer MATCH_GROUP_EXPRESSION = 2;
+
+    public RawExpression(String rawExpression) {
+        if (rawExpression == null || rawExpression.isEmpty()) {
+            throw new IllegalArgumentException(ErrorMessage.INVALID_RAW_EXPRESSION);
+        }
+        this.expression = rawExpression;
+        Matcher matcher = Pattern
+                .compile(CUSTOM_DELIMITER_REGEX)
```

Comment:
현재 메서드 호출할때마다 Pattern.compile()을 호출하고 있네요 :)
Pattern.compile 내부 구현을 보면
```
// Pattern
public static Pattern compile(String regex) {
    return new Pattern(regex, 0);
}
```
매번 메서드 호출시 성능이 비싼 Pattern 인스턴스를 생성하고 있습니다.
Pattern 인스턴스는 한번 쓰고 버려지기 때문에 성능적으로 좋지 않습니다. 
자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다!
```
If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time
```

불필요한 객체 생성을 피하도록 개선해주시면 좋을 것 같아요 😄

Reply:
앗 감사합니다~ㅎㅎ